### ISSUE TITLE
Pipeline Sanity check

### ISSUE THREAD

**Issue by GabrieleMeoni** [GabrieleMeoni/END2END#1]
* Check labels vs images:
 -- Check images are opened as expected
 -- Check labels are categorical and matching the images
- Check all of this at the output of dataloader
* Check image normalization 
* Check expected prediction
-- Check image prediction is as expected
-- Check prediction is compared to the right label
* Check the models weight are updated correctly
* Check losses
* Check learning rate
